isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/backend/anwendungskern.adoc

```diff
@@ -0,0 +1,201 @@
+.Namenskonvention Anwendungsfunktion
+[cols="1s,4m",options="header"]
+|====
+2+|Anwendungsfunktion
+|Schema |Afu<Anwendungsfunktion>
+|Beispiele |AfuBerechnungFristdatum
+
+AfuErmittlungEntscheidungsrelevanz
+|====
+
+// end::namenskonvention[]
+
+Die folgende Abbildung fasst den <<image-aufbau-fachkomponente>> noch einmal grafisch zusammen.
+
+[[image-aufbau-fachkomponente]]
+.Aufbau einer Fachkomponente
+image::software-technisch/backend/aufbau_fachkomponente_awk.dn.svg[]
+
+[[das-anwendungskern-framework]]
+== Anwendungskern-Framework
+
+// tag::awk-framework-definition[]
+
+Für querschnittliche Funktionalität innerhalb des Anwendungskerns wird das Spring-Framework genutzt.
+Hauptaufgabe des Frameworks ist es, die Komponenten zu konfigurieren und miteinander bekanntzumachen.
+Dadurch wird die Trennung zwischen Fachlichkeit und Technik verbessert.
+Beispiel für querschnittliche Funktionalität ist die deklarative Steuerung von Transaktionen.
+
+// end::awk-framework-definition[]
+
+Allgemeine Vorgaben hierzu sind in der xref:software-technisch/backend/spring.adoc[] beschrieben.
+
+[[konfiguration-fachkomponente]]
+=== Konfiguration von Fachkomponenten
+
+Fachkomponenten werden durchgängig mittels Spring konfiguriert.
+Das umfasst vor allem, aber nicht ausschließlich, die Fassade (Schnittstellen der Fachkomponente) und zugehörige Anwendungsfallklassen.
+
+Bei Anwendungsfunktionsklassen oder Hilfsklassen ist je nach Relevanz zu entscheiden, ob diese als eigene Spring Beans definiert werden.
+Im Zweifel sollte die Konfiguration über Spring bevorzugt werden.
+Wenn eine Klasse nur an einer Stelle genutzt wird, kann sie als Kompromiss auch als anonyme Spring Bean definiert werden.
+Sind Klassen nicht von relevanter Bedeutung, so können sie beim Erzeugen der Spring Bean programmatisch erzeugt werden.
+
+[[service-consumer]]
+== Service Consumer
+
+// tag::service-consumer-definition[]
+
+Wenn Backends Services anderer IT-Systeme zur Erfüllung ihrer fachlichen Aufgaben benötigen, so werden diese als Komponente ("Service Consumer") im Anwendungskern abgebildet.
+Dabei ist es für ein Backend transparent, ob der Service innerhalb oder außerhalb der Anwendungslandschaft beheimatet ist.
+Service Consumer verhalten sich wie Fachkomponenten des Anwendungskerns.
+Sie übernehmen die technische Kommunikation mit dem Service sowie die Abbildung von Transportobjekten des Services auf Geschäftsobjekte und Exceptions des Backends.
+Dadurch ist ihre Funktionalität sauber gekapselt, sodass technische Änderungen am Service im Idealfall nur zu Änderungen am Service Consumer führen.
+
+// end::service-consumer-definition[]
+
+[[transaktionssteuerung]]
+== Transaktionssteuerung
+
+An der Schnittstelle des Anwendungskerns findet die Transaktionssteuerung statt.
+Jeder Aufruf wird in einer eigenen Transaktion abgearbeitet.
+Die Transaktion wird beim Aufruf des Anwendungskerns gestartet und mit der Rückgabe des Ergebnisses abgeschlossen (Commit) und somit beendet.
```

Comment:
Für Batches kann es sinnvoll sein, transaktionslose Schnittstellen bereit zu stellen, (Langläufer-Methoden), 
die ihrerseits in einer großen Schleife immer wieder neue kurze Transaktionen öffnen und committen

Reply:
Wenn ihr mit Transaction-Attribute "Supported" auskommt, alles klar, dann bleibt das natürlich so